### PR TITLE
added filters to get orders

### PR DESCRIPTION
**Pull request by nitzantomer**

### CODE REVIEW

File: scripts/src/models/orders.ts [hunk 1]
```diff
@@ -32,13 +32,32 @@ function updateQueryWithStatus(query: SelectQueryBuilder<any>, status?: OrderSta
...
 export type OrderStatic<T extends Order = Order> = {
 	CLASS_ORIGIN: OrderOrigin | null;
 
 	new(): T;
 	createQueryBuilder(): SelectQueryBuilder<BaseEntity>;
 };
 
+export type GetOrderFilters = {
+	userId: string;
+	origin?: string;
```
Comment (doodyparizada) [inline scripts/src/models/orders.ts]: OrderOrigin

File: scripts/src/models/orders.ts [hunk 2]
```diff
@@ -99,21 +118,25 @@ export class Order extends CreationDateModel {
...
-	public static getAll<T extends Order>(this: OrderStatic<T> | Function, userId: string, limit: number): Promise<T[]>;
+	/*public static getAll<T extends Order>(this: OrderStatic<T> | Function, userId: string, limit: number): Promise<T[]>;
 	public static getAll<T extends Order>(this: OrderStatic<T> | Function, userId: string, status: OrderStatusAndNegation): Promise<T[]>;
+	public static getAll<T extends Order>(this: OrderStatic<T> | Function, userId: string): Promise<T[]>;
 	public static getAll<T extends Order>(this: OrderStatic<T> | Function, userId: string, status: OrderStatusAndNegation, limit: number): Promise<T[]>;
-	public static getAll<T extends Order>(this: OrderStatic<T> | Function, userId: string, second?: number | OrderStatusAndNegation, third?: number): Promise<T[]> {
-		const status: OrderStatusAndNegation | null = typeof second === "string" ? second : null;
-		const limit: number | null = typeof second === "number" ? second : (typeof third === "number" ? third : null);
+	public static getAll<T extends Order>(this: OrderStatic<T> | Function, userId: string, second?: number | OrderStatusAndNegation, third?: number): Promise<T[]> {*/
```
Comment (doodyparizada) [inline scripts/src/models/orders.ts]: why so much commented out code

File: scripts/src/models/orders.ts [hunk 3]
```diff
@@ -32,13 +32,32 @@ function updateQueryWithStatus(query: SelectQueryBuilder<any>, status?: OrderSta
...
+
 export type OrderStatic<T extends Order = Order> = {
 	CLASS_ORIGIN: OrderOrigin | null;
 
 	new(): T;
 	createQueryBuilder(): SelectQueryBuilder<BaseEntity>;
 };
 
+export type GetOrderFilters = {
+	userId: string;
```
Comment (doodyparizada) [inline scripts/src/models/orders.ts]: you can never request other users history

File: scripts/src/public/routes/orders.ts [hunk 4]
```diff
@@ -100,11 +100,18 @@ export const cancelOrder = async function(req: Request, res: Response) {
 	res.status(204).send();
 } as any as RequestHandler;
 
+export type GetOrderHistoryRequest = Request & {
+	query: {
+		origin?: string;
+		offerId?: string;
```
Comment (doodyparizada) [inline scripts/src/public/routes/orders.ts]: offer_id

File: tests/src/services/orders.spec.ts [hunk 5]
```diff
@@ -30,7 +30,7 @@ describe("test orders", async () => {
 	test("getAllNonOpen", async () => {
 		const user = await helpers.createUser();
 		await helpers.createOrders(user.id);
-		const orders = await Order.getAll(user.id, "!opened", 25);
+		const orders = await Order.getAll({ userId: user.id, status: "!opened" }, 25);
```
Comment (doodyparizada) [inline tests/src/services/orders.spec.ts]: can you add more tests? specifically using the new filters?

File: scripts/src/models/orders.ts [hunk 6]
```diff
@@ -99,21 +118,18 @@ export class Order extends CreationDateModel {
...
 			.addOrderBy("id", "DESC");
 
-		updateQueryWithStatus(query, status);
+		// updateQueryWithStatus(query, filters.status);
+		updateQueryWithFilter(query, "status", filters.status);
+		updateQueryWithFilter(query, "origin", filters.origin);
+		updateQueryWithFilter(query, "offer_id", filters.offerId);
 
-		if ((this as OrderStatic<T>).CLASS_ORIGIN) {
+		if (!filters.origin && (this as OrderStatic<T>).CLASS_ORIGIN) {
```
Comment (doodyparizada) [inline scripts/src/models/orders.ts]: I would test this functionality + document whats going on here

File: scripts/src/utils.ts [hunk 7]
```diff
@@ -14,10 +19,25 @@ export function path(...paths: string[]): string {
...
 export function random(min: number, max: number): number;
-export function random(min?: number, max?: number): number {
-	if (min !== undefined && max !== undefined) {
-		return Math.random() * (max - min) + min;
+export function random(first?: number | Map<string, any> | SimpleObject | any[], second?: number): number | [string, any] | any {
+	if (first instanceof Map) {
+		first = Array.from(first.entries());
+	}
+
+	if (isSimpleObject(first)) {
```
Comment (doodyparizada) [inline scripts/src/utils.ts]: don't you mean `else if`?

File: scripts/src/utils.ts [hunk 8]
```diff
@@ -14,10 +19,25 @@ export function path(...paths: string[]): string {
...
+		first = Object.keys(first).map(key => [key, (first as SimpleObject)[key]]);
+	}
+
+	if (Array.isArray(first)) {
```
Comment (doodyparizada) [inline scripts/src/utils.ts]: else if?

Comment (nitzantomer) [reply scripts/src/utils.ts]: no, not here.
because in the firs two `if`s I actually replace `first` with an array

File: scripts/src/utils.ts [hunk 9]
```diff
@@ -14,10 +19,25 @@ export function path(...paths: string[]): string {
...
+		return first[Math.floor(Math.random() * first.length)];
+	}
+
+	if (first !== undefined && second !== undefined) {
+		return Math.random() * (second - (first as number)) + (first as number);
 	}
```
Comment (doodyparizada) [inline scripts/src/utils.ts]: I'd separate this function into different functions and pay with duplicate code as it's flow isn't that clear for a simple random function.

Comment (nitzantomer) [reply scripts/src/utils.ts]: i like it that i can use `random([1, 2, 3])` and not something like `randomItem(...)`

**Pull request merged**